web/src/components/upload/UploadFormCard.vue

```diff
@@ -0,0 +1,494 @@
+          },
+        }
+      );
+
+      // Create a new report object and store it in the reports list.
+      // This will be persisted in local storage.
+      reports.value.push({
+        id: response.data["id"],
+        name: name.value ?? "",
+        date: new Date().toISOString(),
+        status: response.data["status"],
+        statusUrl: response.data["url"],
+      });
+
+      // Set the report as active.
+      reportActiveId.value = response.data["id"];
+
+      // Make sure a status url was provided.
+      if (!reportActive.value?.statusUrl) {
+        handleError("No analysis URL was provided by the API.");
+      }
+    } catch (e: any) {
+      if (e.code == "ERR_NETWORK") {
+        handleError("Could not connect to the API.");
+      } else {
+        handleError("An error occurred while uploading the file: " + e.message);
+      }
+    } finally {
+      uploadProgress.value = 0;
+    }
+  }
+};
+
+// Poll configuration
+const pollingInterval = 1000;
+const pollingMax = 60;
+
+// List containing the the ids of the reports that are currently polling.
+const pollingReports = ref<string[]>([]);
+
+// Start polling for a specific report.
+const startPolling = (reportId: string): void => {
+  // Do not start polling when the report is already polling.
+  if (pollingReports.value.includes(reportId)) {
+    return;
+  }
+
+  // Create the interval for polling.
+  const interval = setInterval(async () => {
+    // Stop the interval when the report is not in the list of polling reports.
+    if (!pollingReports.value.includes(reportId)) {
+      clearInterval(interval);
+      return;
+    }
+
+    // Get the report from the reports list.
+    const report = reports.value.find((r) => r.id === reportId);
+
+    // Stop the polling if the report no longer exists.
+    if (!report) {
+      clearInterval(interval);
+      return;
+    }
+
+    try {
+      const response = await axios.get(report.statusUrl);
+
+      // Update the report status.
+      report.status = response.data.status;
+      // Update the report response.
+      report.response = response.data;
+
+      // Stop the polling when the report status is final.
+      if (
+        report.status === "finished" ||
+        report.status === "error" ||
+        report.status === "failed"
+      ) {
+        delete pollingReports[report.id];
+        clearInterval(interval);
+      }
+
+      // If the report is the active report
+      // apply some changes to the form UI.
+      if (report === reportActive.value) {
+        if (report.status === "finished") {
+          // Go to the results page.
+          step.value = 4;
+          // Clear the form.
+          clearForm();
+        }
+
+        if (report.status === "error" || report.status === "failed") {
+          handleError(response.data.error);
+        }
+      }
+    } catch (e: any) {
+      //
+    }
+  }, pollingInterval);
+
+  // Add the report.
+  pollingReports.value.push(reportId);
+};
```

Comment:
You've already split the original file up into multiple parts, which is good. But this component is still very large.

Would it make sense to put some of the logic here in a store?

Reply:
This was later implemented in the "report store"